src/Symfony/Component/Notifier/Bridge/KazInfoTeh/KazInfoTehTransport.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\KazInfoTeh;
+
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+
+/**
+ * @author Egor Taranov <dev@taranovegor.com>
+ */
+class KazInfoTehTransport extends AbstractTransport
+{
+    protected const HOST = 'kazinfoteh.org';
+
+    /**
+     * @var string
+     */
+    private $username;
+
+    /**
+     * @var string
+     */
+    private $password;
+
+    /**
+     * @var string
+     */
+    private $sender;
+
+    public function __construct(string $username, string $password, string $sender, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->username = $username;
+        $this->password = $password;
+        $this->sender = $sender;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf(
+            'kazinfoteh://%s?sender=%s',
+            $this->getEndpoint(),
+            $this->sender
+        );
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage
+            && 0 === strpos($message->getPhone(), '77')
+            && 11 === strlen($message->getPhone())
+        ;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$this->supports($message)) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = sprintf('http://%s/api', $this->getEndpoint());
```

Comment:
This is not good, but there is no other solution. The connection is made via the http protocol and port 9507.

Reply:
This looks good to me